content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,126 @@
+2. Rebasing no longer has to rebuild everything. As the standard library is now built *after* the compiler instead of before, all dependencies and compiler crates that are unmodified can be cached when switching `git` branches. It's only necessarily to rebuild everything after bootstrap bumps every 6 weeks as part of the release cycle.
+3. Modifying the standard library no longer has to rebuild everything. After the redesign, it is now possible to add documentation comments to functions in the standard library without having to rebuild the compiler and then also rebuild std a second time. `--keep-stage-std=0` is no longer needed.
+4. It aligns better with how other rust programs are built. We no longer have a "strange" setup where the stage 1 compiler was built from a *beta* rustc with an *in-tree* std. Now, the stage 1 compiler is built from a beta rustc and a beta std.
+
+# In terms of bootstrap invocations and bootstrap config, what does this redesign mean?
+
+The minimum stage to check, build and test the standard library is now stage 1. `./x {check,build,test} library --stage=0` are now no-ops; switch to `--stage 1` instead. `--keep-stage-std=0` is a no-op.
+
+For `profile = "library"` users, like aforementioned, the default check, build, and test stage are now bumped to 1. `download-rustc = "if-unchanged"` is enabled by default, which downloads a pre-built CI rustc instead of building the compiler if there are no compiler changes, allowing you to build the standard library without building the compiler.
+
+# What does this mean for contributors working on the standard library and the compiler?
+
+- Contributors will now no longer need to use `cfg(bootstrap)` for intrinsics and lang items.
+- Contributors may (rarely) need to use `cfg(bootstrap)` in compiler code if they wish to experiment with unstable library features [^dogfood-unstable-lib-features]
+
+# Frequently asked questions (FAQs) {#faqs}
+
+## Doesn't this just shift `cfg(bootstrap)` from library code to compiler code? {#faqs-shift-cfg-bootstrap}
+
+Not quite. `cfg(bootstrap)` usage in standard library code for using new intrinsics / lang items (as in the current bootstrap sequence) is much more common than potential `cfg(bootstrap)` usage in compiler code for experimenting with unstable library features (as in the redesigned bootstrap sequence). This is because the standard library need to depend on compiler-provided lang items and intrinsics, but the compiler does not (need to) depend on standard library implementation details.
+
+Additionally, the compiler only needs to add `cfg(bootstrap)` for unstable library features *not yet on beta*. By our count, the compiler has not used a feature added that recently since before 1.61.
```

Comment:
"not yet on beta" feels a bit misleading here -- what it really wants to say is anything in the standard library that has changed unstable API and is used in the compiler.

(Out of interest, how did we perform this analysis?)

Reply:
Right.

> (Out of interest, how did we perform this analysis?)

I can't claim to have done this analysis myself (though it *seems* reasonably accurate to  me). I believe @jyn514 did the analysis originally?

Reply:
> I can't claim to have done this analysis myself (though it _seems_ reasonably accurate to me). I believe @jyn514 did the analysis originally?

see https://jyn.dev/bootstrapping-rust-in-2023/#what-can-we-do-about-it and in particular https://github.com/jyn514/rust/tree/versions-used